### PR TITLE
fix: page.pause() timeout reset

### PR DESCRIPTION
**Pull request by mxschmitt**
Fixes https://github.com/microsoft/playwright-python/issues/2154

### CODE REVIEW

File: playwright/_impl/_browser_context.py [hunk 1]
```diff
@@ -248,15 +248,18 @@ def set_default_navigation_timeout(self, timeout: float) -> None:
     def _set_default_navigation_timeout_impl(self, timeout: Optional[float]) -> None:
         self._timeout_settings.set_default_navigation_timeout(timeout)
         self._channel.send_no_reply(
-            "setDefaultNavigationTimeoutNoReply", dict(timeout=timeout)
+            "setDefaultNavigationTimeoutNoReply",
+            {} if timeout is None else {"timeout": timeout},
```
Comment (mxschmitt) [inline playwright/_impl/_browser_context.py:252]: The issue was that on protocol level we don't allow timeout to be null.

File: tests/async/test_page.py [hunk 2]
```diff
@@ -1294,3 +1294,33 @@ def binding(source, o):
...
+
+async def test_page_pause_should_reset_default_timeouts(
+    page: Page, headless: bool, server: Server
+) -> None:
+    if not headless:
+        pytest.skip()
+
+    await page.goto(server.EMPTY_PAGE)
+    page.pause()
+    with pytest.raises(Error, match="Timeout 30000ms exceeded."):
```
Comment (yury-s) [inline tests/async/test_page.py:1307]: Do we really want to slow down all tests by 30s?

File: playwright/_impl/_browser_context.py (hunk 1 above)
Comment (yury-s) [reply playwright/_impl/_browser_context.py:252]: Should we consider serializing None as undefined or too many things break?

**Review by yury-s — APPROVED** [review]

File: playwright/_impl/_browser_context.py (hunk 1 above)
Comment (mxschmitt) [reply playwright/_impl/_browser_context.py:252]: too many things would break I believe!

File: tests/async/test_page.py (hunk 2 above)
Comment (mxschmitt) [reply tests/async/test_page.py:1307]: I think no other option instead of hacking around the impl for testing.

**Pull request merged**